project/src/main/puzzle/puzzle-state.gd

```diff
@@ -112,20 +117,56 @@ func _physics_process(_delta: float) -> void:
 		input_frame += 1
 
 
+## Creates and starts a one-shot timer.
+##
+## This timer is freed when it times out or when the puzzle is interrupted.
+##
+## Parameters:
+## 	'wait_time': The amount of time to wait. A value of '0.0' will result in an error.
+##
+## Returns:
+## 	A timer which has been added to the scene tree, and is currently active.
+func start_timer(wait_time: float) -> Timer:
+	var timer := add_timer(wait_time)
+	timer.start()
+	return timer
+
+
+## Creates a one-shot timer, but does not start it.
+##
+## This timer is freed when it times out or when the puzzle is interrupted.
+##
+## Parameters:
+## 	'wait_time': The amount of time to wait. A value of '0.0' will result in an error.
+##
+## Returns:
+## 	A timer which has been added to the scene tree, but is not yet active.
+func add_timer(wait_time: float) -> Timer:
+	var timer := Timer.new()
+	timer.one_shot = true
+	timer.wait_time = wait_time
+	timer.connect("timeout", self, "_on_Timer_timeout_queue_free", [timer])
+	_timers.add_child(timer)
+	return timer
+
+
+## Cleans up any temporary timers and listeners.
```

Comment:
Rename this method, it only cleans up timers.

Reply:
I prefer the current name; all timer have associated listeners, and it's also symmetrical with the tutorial method of the same name.